src/algorithms/policy_gradient/pg.jl

```diff
@@ -0,0 +1,63 @@
+using Random
+using Flux
+
+using ReinforcementLearningBase
+using ReinforcementLearningCore
+using Distributions
+
+export PGPolicy
+Base.@kwdef mutable struct PGPolicy{A<:NeuralNetworkApproximator,R<:AbstractRNG} <:
+                           AbstractPolicy
+    approximator::A
+    dist_fn::Any = Normal # distribution function
+    γ::Float32 = 0.99f0 # discount factor
+    rng::R = Random.GLOBAL_RNG
+    loss::Float32 = 0f0
+end
+
+# if the action space is discrete, then softmax should be the last layer of the model.
+
+function (π::PGPolicy)(env)
+    to_dev = x -> send_to_device(device(π.approximator), x)
+
+    logits = env |> get_state |> to_dev |> π.approximator |> send_to_host # if the action space is discrete, then the approximator should softmax the output values.
```

Comment:
I think we can remove the `softmax` constraint of `approximator` here. So that we can use `logsoftmax` when updating to avoid some strange corner cases. If so, the test case needs to be updated.